unified-scheduler-pool/src/lib.rs

```diff
@@ -603,6 +721,19 @@ impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
                 };
 
                 loop {
+                    if let Ok(NewTaskPayload::OpenSubchannel(context)) = new_task_receiver.recv() {
+                        // signal about new SchedulingContext to handler threads
+                        runnable_task_sender
+                            .send_chained_channel(context, handler_count)
+                            .unwrap();
+                        assert_matches!(
+                            result_with_timings.replace(initialized_result_with_timings()),
+                            None
+                        );
+                    } else {
+                        unreachable!();
```

Comment:
like bunch of `.unwrap()`s here and there, this `unreachable!()` will be replaced with proper thread shutdown code.